lib/vector-core/src/schema/definition.rs

```diff
@@ -161,14 +203,6 @@ impl Definition {
     pub fn with_known_meaning(mut self, path: impl Into<LookupBuf>, meaning: &str) -> Self {
         let path = path.into();
 
-        // Ensure the path exists in the collection.
-        assert!(self
-            .collection
-            .find_known_at_path(&mut path.to_lookup())
-            .ok()
-            .flatten()
-            .is_some());
-
```

Comment:
The concept of "optional" is now fully baked into the type definition. Previously it was storing a separate set of fields, and right before remap it would add `null` back onto the field. (see https://github.com/vectordotdev/vector/pull/13103)

I think the `find_known_at_path` concept was a mistake. (see https://github.com/vectordotdev/vector/issues/13045). This should be an internal detail of the type definition itself. When using it, all you should be able to ask is , "what is the type of this path", which includes the "unknown". That, along with the fact that optional is just a type that can be `null` means this assertion is useless. You can theoretically have a semantic meaning pointing to a path that isn't a "known" path (for example an optional field of JSON that is part of the "unknown" type).

Reply:
actually, thinking about it more we still need this check when the `unknown` is empty, so I'll add this back, but use `find_at_path` instead.